### PR TITLE
Include manifest creation in application

### PR DESCRIPTION
**Pull request by wow-such-code**
- Adds manifest creation use case
- Adds view components to call that use case and display its results
- Adds Controller and sets up the manifest creation use case and view components in the dependency manager

### CODE REVIEW

**Comment by wow-such-code** [comment]
[create a manifest for the given sample ids and allow the user to download it to the users system](https://trello.com/c/vsAMDw7e/939-create-a-manifest-for-the-given-sample-ids-and-allow-the-user-to-download-it-to-the-users-system)

File: sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/DependencyManager.groovy [hunk 1]
```diff
@@ -147,6 +162,27 @@ class DependencyManager {
...
+        //CreateAffiliation createAffiliation = new CreateAffiliation(createAffiliationPresenter, dataSource)
+        //controller -> use case
+        //CreateAffiliationController createAffiliationController = new CreateAffiliationController(createAffiliation)
+        //View -> Controller, Model
+        //return new CreateAffiliationView(sharedViewModel, createAffiliationViewModel, createAffiliationController)
+        
+        //model -> presenter -> use case -> controller -> view
+        //model -> (presenter -> use case2) -> use case -> controller -> view
+    }
+*/
```
Comment (KochTobi) [inline sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/DependencyManager.groovy:185]:
```suggestion
```
We can remove this @wow-such-code, correct?

**Review by jenniferboedker — COMMENTED** [review]
I think it would be nice if the button is disabled if no project is selected.

Also, is it possible to close the notification when clicking the screen at any place? I think it's not necessary to use such a prominent notification.

I will have a look at the code now.

File: sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/DependencyManager.groovy [hunk 2]
```diff
@@ -147,6 +162,27 @@ class DependencyManager {
...
+        //CreateAffiliationOutput createAffiliationPresenter = new CreateAffiliationPresenter(sharedViewModel, createAffiliationViewModel)
+        //use case -> datasource? output
+        //CreateAffiliation createAffiliation = new CreateAffiliation(createAffiliationPresenter, dataSource)
+        //controller -> use case
+        //CreateAffiliationController createAffiliationController = new CreateAffiliationController(createAffiliation)
+        //View -> Controller, Model
+        //return new CreateAffiliationView(sharedViewModel, createAffiliationViewModel, createAffiliationController)
+        
+        //model -> presenter -> use case -> controller -> view
+        //model -> (presenter -> use case2) -> use case -> controller -> view
```
[the remark this answers is not in the document]
Comment (wow-such-code) [reply sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/DependencyManager.groovy]: To help me learn how to connect these components. :) Have been removed now.

File: sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewView.groovy [hunk 3]
```diff
@@ -18,29 +18,59 @@ import com.vaadin.ui.themes.ValoTheme
...
+    private void setupProjects(){
         projectGrid = new Grid<>()
+        fillProjectsGrid()
+        projectGrid.setSelectionMode(Grid.SelectionMode.SINGLE)
+        projectGrid.addSelectionListener({
+            if (it instanceof SingleSelectionEvent<ProjectSummary>) {
+                clearProjectSelection()
+                it.getSelectedItem().ifPresent(this::selectProject)
+            }
+        })
```
[the remark this answers is not in the document]
Comment (KochTobi) [reply sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewView.groovy:57]: I will take care of this

File: sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewView.groovy [hunk 4]
```diff
@@ -18,29 +18,59 @@ import com.vaadin.ui.themes.ValoTheme
 */
 class ProjectOverviewView extends VerticalLayout{
 
+    private final ProjectOverviewViewModel viewModel
+    private final DownloadProjectController downloadProjectController
+
     private Label titleLabel
-    private ProjectOverviewViewModel viewModel
     private Grid<ProjectSummary> projectGrid
+    private TextArea manifestArea
```
[the remark this answers is not in the document]
Comment (jenniferboedker) [reply sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewView.groovy:26]: maybe you can add todos that hint what needs to be removed when the file is finished :)

[the remark this answers is not in the document]
Comment (KochTobi) [reply sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewView.groovy:26]: I think that this should not be part of this PR but a separate PR. Adding TODOs without tackling them can be confusing

File: sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewView.groovy [hunk 5]
```diff
@@ -18,29 +18,59 @@ import com.vaadin.ui.themes.ValoTheme
...
+    }
+
+    private void setupProjects(){
         projectGrid = new Grid<>()
+        fillProjectsGrid()
+        projectGrid.setSelectionMode(Grid.SelectionMode.SINGLE)
+        projectGrid.addSelectionListener({
+            if (it instanceof SingleSelectionEvent<ProjectSummary>) {
+                clearProjectSelection()
+                it.getSelectedItem().ifPresent(this::selectProject)
```
Comment (jenniferboedker) [inline sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewView.groovy:55]: What is selectProject?

File: sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewView.groovy (hunk 4 above)
[the remark this answers is not in the document]
Comment (jenniferboedker) [reply sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewView.groovy:26]: But you added the code here with the knowledge it should be removed.

File: sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewView.groovy [hunk 6]
```diff
@@ -65,4 +95,25 @@ class ProjectOverviewView extends VerticalLayout{
...
+        viewModel.addPropertyChangeListener("generatedManifest", {
+            textArea.setValue(Optional.ofNullable(it.newValue).orElse("") as String)
+        })
+        return textArea
+    }
+
+    private AbstractComponent setupProjectSpecificButtons() {
+
+        MenuBar buttonBar = new MenuBar()
+        buttonBar.addItem("Download Project", {
```
Comment (jenniferboedker) [inline sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewView.groovy:112]: You do not download the project but the manifest file for the samples. Maybe just call it Download Data or Download Manifest. (You can download data when the project is not completed too)

**Comment by jenniferboedker** [comment]
I am missing the download here. Why do you not download a file to the user's system. When should the file be created?

File: sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewView.groovy (hunk 5 above)
Comment (KochTobi) [reply sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewView.groovy:55]: > I am missing the download here. Why do you not download a file to the user's system. When should the file be created?

The download will be handled in a separate PR

**Comment by KochTobi** [comment]
> I am missing the download here. Why do you not download a file to the user's system. When should the file be created?

The download will be handled in a separate PR

File: sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewView.groovy (hunk 6 above)
Comment (wow-such-code) [reply sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewView.groovy]: Called it "Get Available Data", also good?

**Review by jenniferboedker — CHANGES_REQUESTED** [review]
I would prefer another name for the button to be as precise as possible. Furthermore, please add a description of what to review here because as it seems some stuff should not be reviewed since it is the work of tobi. On what are you focusing with this PR?

**Review by jenniferboedker — APPROVED** [review]

**Pull request merged**